plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/inherit-query-control.tsx

```diff
@@ -128,4 +87,41 @@ const InheritQueryControl = ( {
 	);
 };
 
-export default InheritQueryControl;
+const FilterableControl = ( {
+	setQueryAttribute,
+	trackInteraction,
+	query,
+}: QueryControlProps ) => {
+	const filterable = query?.filterable;
+
+	const defaultValue = useMemo( () => getDefaultValueOfFilterable(), [] );
+
+	return (
+		<ToolsPanelItem
+			label={ label }
+			hasValue={ () => filterable !== defaultValue }
+			isShownByDefault
+			onDeselect={ () => {
+				setQueryAttribute( {
+					filterable: defaultValue,
+				} );
+				trackInteraction( CoreFilterNames.FILTERABLE );
+			} }
+		>
+			<ToggleControl
+				className="wc-block-product-collection__inherit-query-control"
```

Comment:
Class name need to be updated:

```suggestion
				className="wc-block-product-collection__filterable-control"
```

Reply:
I was thinking about that and eventually left the same `className` on purpose. From UI POV it's exactly the same control, hence I thought we should keep the same class. Maybe it should be renamed according to UI purposes, for example: `wc-block-product-collection__use-page-context-control`. And in such case, the same class could be used in both cases. WDYT?

Reply:
Aha! Got it. That makes sense to me 🤝

Reply:
Agreed with Manish elsewhere to keep it as-is. I renamed the whole file though to `use-page-context-control`.